code/tasks/UserFormsUpgradeService.php

```diff
@@ -224,4 +230,66 @@ public function getQuiet()
     {
         return $this->quiet;
     }
+
+
+    /**
+     * Migrate Placeholder data from field specific table to the EditableFormField table
+     *
+     * @param EditableFormField $field
+     * @param string $tableName
+     */
+    private function migratePlaceholder($field, $tableName)
+    {
+        // Migrate Placeholder setting from $tableName table to EditableFormField table
+        if ($field->Placeholder) {
+            return;
+        }
+        // Check if draft table exists
+        $query = "SHOW TABLES LIKE '$tableName'";
```

Comment:
also `DB::get_schema()->hasTable($tableName)` for shorthand.

also `DB::get_schema()->hasField($tableName, $fieldName)` to check both at the same time :)